[14.0][NEW] l10n_br_pos_cfe

ygcarvalh opened the pull request:
Baseado no módulo da localização para o Ponto de Venda, adiciona as funcionalidades necessárias para a emissão do CFe ao realizar o fluxo pelo frontend.

Depende do PR https://github.com/OCA/l10n-brazil/pull/2300.

Esse PR foi criado a partir do PR https://github.com/OCA/l10n-brazil/pull/2096.

cc: @mileo @rvalyi @marcelsavegnago @lfdivino @luismalta

mileo committed:
[NEW] l10n_br_pos_cfe module scaffold

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[REF] chrome.js to l10n_br_pos_cfe

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[REF] Extract cfe fields to l10n_br_pos_cfe

mileo committed:
[MOV] CF-E fields to l10n_br_pos_cfe

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[MOV] JS SAT device to l10n_br_pos_sat

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[IMP] Fiscal CF-e device

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[REF] l10n_br_pos_cfe models

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[IMP] cf-e init

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[FIX] print sat

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[REF] CF-e response processing

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[IMP] Last session number

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[REF] CFe key

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[FIX] Cancellation pop-up

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[REF] Clean code of l10n_br_pos_cfe

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[IMP] Cancelamento e status da CF-e

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[NEW] README files

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

ygcarvalh committed:
[IMP] CFe printing via QWeb

mileo committed:
[IMP] Pos SAT demo configuration

ygcarvalh committed:
[IMP] Overwrite the original template for OrderReceipt

ygcarvalh committed:
[IMP] Missing information for CFe printing via QWeb

luismalta committed:
[FIX] Paymentline forEach key in receipt template

mileo committed:
[REF] remove <data> from XML

ygcarvalh committed:
[IMP] CFe cancellation flow

luismalta committed:
[REF] Remove optional chain syntax

Signed-off-by: Luis Malta <luis.malta@kmee.com.br>

ygcarvalh committed:
[FIX] Information from the fields that are sent to the CFe

lfdivino committed:
[FIX] Remove non numbers at client document

ygcarvalh committed:
[REM] Dead/unusable code removal

ygcarvalh committed:
[IMP] Implementation of POS flow tests

ygcarvalh committed:
[REM] Code with console and some unused functions

ygcarvalh committed:
[IMP] Add contributors to the addons readme

ygcarvalh committed:
[IMP] New unit tests

ygcarvalh committed:
[ADD] pre-commit setup

mileo commented:
@ygcarvalh @luismalta @lfdivino podem fazer o rebase?

ygcarvalh committed:
[REM] l10n_br_pos_cfe: Remove data files for payment method

ygcarvalh committed:
[FIX] l10n_br_pos_cfe: Change the variable names from en to pt

rvalyi commented on l10n_br_pos_cfe/__manifest__.py:
```diff
@@ -0,0 +1,41 @@
+    "website": "https://github.com/OCA/l10n-brazil",
+    "category": "Point Of Sale",
+    "development_status": "Alpha",
+    "maintainers": ["mileo", "sadamo", "gabrielcardoso21", "lfdivino"],
```
mesma observação que eu tinha feito no l10n_br_pos: sadamo e gabrielcardoso21 são apenas autores do modulo ou pretendem manter de forma ativa? Pois estão mais de 2 anos sem fazer nada na OCA...

rvalyi commented on l10n_br_pos_cfe/readme/INSTALL.rst:
```diff
@@ -0,0 +1,8 @@
+
+* l10n_br_pos
+
+Alem disso você vi precisar de um IOT BOX, podendo ser o da Odoo SA ou o Pywebdriver, com os respectivos drivers para CF-E:
```
substituir:
s/vi/vai/

rvalyi commented on l10n_br_pos_cfe/static/src/xml/Screens/ReceiptScreen/OrderSubtitleReceipt.xml:
```diff
@@ -0,0 +1,36 @@
+                    > >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> </span>
+                    <span
+                        class="test-env__divider"
+                    > >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> </span>
```
esses caracteres >>>> estão aqui de proposito? (para que então?)

mileo replied:
Estão sim, fazem parte do cupom emitido em homologação.

ygcarvalh replied:
Isso mesmo. É uma das necessidades que a documentação pede nesses casos.

mileo commented on l10n_br_pos_cfe/README.rst:
```diff
@@ -0,0 +1,148 @@
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+Integração com equipamento CF-E para emissão de Cupom Fiscal.
```
@ygcarvalh pode adicionar na descrição que o módulo funciona para os equipamentos SAT (SP) e MF-E (Ceará)

ygcarvalh committed:
[IMP] l10n_br_pos_cfe: update readme and manifest files

Removal of some maintainers who are no longer actively participating in
the community; addition of supported equipment for this module and
generation of the readme with the changes made.

mileo commented on l10n_br_pos_cfe/models/pos_payment_method.py:
```diff
@@ -0,0 +1,69 @@
+    _inherit = "pos.payment.method"
+
+    sat_payment_mode = fields.Selection(WA03_CMP_MP, "SAT Payment Mode")
+    sat_card_accrediting = fields.Selection(CARD_ACQUIRERS, "Card Acquirer")
```
@ygcarvalh aqui não seria melhor outro nome de variável? 

sat_card_acquirer ?

rvalyi replied:
eu TB acho

antoniospneto reviewed (approved):
lgtm

ygcarvalh committed:
[FIX] l10n_br_pos_cfe: Change sat card var name

Change the card acquirer variable name.

ygcarvalh committed:
[FIX] l10n_br_pos_cfe: update variable name to en

rvalyi commented:
/ocabot merge nobump

ygcarvalh commented:
@rvalyi dei um rebase depois daquela correção do Marcel na `l10n_br_account` só por via das dúvidas 😄

rvalyi commented:
@ygcarvalh a principio o proprio ocabot merge ja faz o rebase (mas é bom mostrar os testes passando mesmo, no caso agora  foi um problema inconsistente com o servidor de IBPT. Não sei porque o ocabot não fez o merge ontem; vamos ver agora).

/ocabot merge nobump
